guides/source/action_controller_response_and_rendering.md

```diff
@@ -0,0 +1,703 @@
+
+- `app/views/book/index.html+mobile.erb`
+- `app/views/book/index.html+desktop.erb`
+- `app/views/book/index.html.erb`
+
+If a template with the specified format does not exist an `ActionView::MissingTemplate` error is raised.
+
+Instead of setting the variant on the render call you may also set it on the request object in your controller action.
+
+```ruby
+def index
+  request.variant = determine_variant
+end
+
+private
+  def determine_variant
+    variant = nil
+    # some code to determine the variant(s) to use
+    variant = :mobile if session[:use_mobile]
+
+    variant
+  end
+```
+
+Creating Responses using `redirect_to`
+-------------------------------------
+
+Another way to create an HTTP response from a controller action is to use the `redirect_to` method. This will send an HTTP redirect response back to the client, instructing the client to send a new request to a different URL.
+
+The `redirect_to` method is typically used to redirect users to a different page after performing some action, such as after submitting a form or after completing a sign-in process.
+
+For example, after creating a book we can `redirect_to` that book's `show` page:
+
+```ruby
+def create
+  @book = Book.new(book_params)
+
+  if @book.save
+    redirect_to book_url(@book), notice: "Book was successfully created."
+  else
+    render :new, status: :unprocessable_entity
+  end
+end
+```
+
+If the `@book` was created successfully we use `redirect_to` to tell the client to make a new HTTP request to that book's show page URL (e.g.`books/42`). If you look at the "network" tab of a browser's dev tools, you will see this second request.
+
+There is also a [`redirect_back`][] method that returns the user to the page
+they just came from. The location is pulled from the `HTTP_REFERER` header.
+Since this header is not guaranteed to be set, you must provide the
+`fallback_location`.
+
+```ruby
+redirect_back(fallback_location: root_path)
+```
+
+NOTE: `redirect_to` and `redirect_back` do not immediately stop the execution of the controller method, they simply set HTTP responses. So statements *after* `redirect_to` in our controller get executed. To terminate the execution of the method immediately after the `redirect_to`, you need to use `return`. For example `redirect_to book_url(@book) and return`
+
+[`redirect_back`]: https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back
+
+### Getting a Different Redirect Status Code
+
+Rails uses HTTP status code 302, a temporary redirect, when you call `redirect_to`. If you'd like to set a different status code, perhaps 301, a permanent redirect, you can use the `:status` option:
+
+```ruby
+redirect_to books_path, status: 301
+```
+
+Just like the `:status` option for `render`, `:status` for `redirect_to` accepts both numeric and symbolic values.
+
+The Difference Between `render` and `redirect_to`
```

Comment:
This section seems to describe when to use `render` and when to use `redirect_to`.
Maybe the following header would be clearer? This would also mean that line 558 should change.
```suggestion
When to use `render` or `redirect_to`
```

Or maybe:
```suggestion
Choosing between `render` and `redirect_to`
```